src/helpers/apify_helper.js

```diff
@@ -85,10 +91,36 @@ async function findDatasetByName(apifyDatasets, maybeDatasetName) {
 
 const randomHostLikeString = () => `${Math.random().toString(36).substring(2)}-${Date.now().toString(36).substring(2)}`;
 
+const uploadInputTable = async (apifyClient, inputFile) => {
+    const { keyValueStores } = apifyClient;
+    const store = await keyValueStores.getOrCreateStore({ storeName: NAME_OF_KEBOOLA_INPUTS_STORE });
+    const storeId = store.id;
+    const key = randomHostLikeString();
+    await keyValueStores.putRecord({
+        storeId,
+        key,
+        body: inputFile,
+        contentType: 'text/csv',
+    });
+    return { storeId, key };
+};
+
+const setRunTimeout = (timeout, runId, actorId) => {
+    setTimeout(async () => {
+        console.log('Extractor timeouts. Saving the state');
+        const stateOutFile = path.join(DATA_DIR, STATE_OUT_FILE);
+        await saveJson(stateOutFile, { runId, actorId });
+        console.log('State saved. Exiting.');
+        process.exit(0);
```

Comment:
Shouldn't it be exit code 1 when it's timeout?

Reply:
0 Should be ok, if there is a state file, the keboola platform knows that it will need to en queue again.